lenskit-core/src/main/java/org/grouplens/lenskit/data/history/History.java

```diff
@@ -75,6 +75,8 @@ private History() {}
      * @return An empty history for the user.
      */
     @Nonnull
+    @SafeVarargs
+    @SuppressWarnings("varargs") // method is safe MDE 2015-05-08
     public static <E extends Event> UserHistory<E> forUser(long id, E... events) {
         List<E> list = ImmutableList.copyOf(events);
```

Comment:
Actually, I think there is another case where it arises.

Since the type of `events` is `Object[]` due to type erasure, non-generic code could call `forUser` with some other type of array (either `Object[]`, `StringBuffer[]`). This would be totally fine and unchecked. We will then have a `List<E>` that contains a string buffer, causing some code down the road to attempt to assign a string buffer to an event and break in code that the type checker proved correct.

I *think* that the correct solution in this case is to suppress the varargs warning (because our code is sound), but remove the `SafeVarargs` annotation so that code *calling* this method might still be analyzed for errors. But I might need to read more on `SafeVarargs` to be sure about that conclusion.